src/resolve/resolved-bus.c

```diff
@@ -1580,6 +1582,132 @@ static int bus_method_reset_server_features(sd_bus_message *message, void *userd
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int bus_method_register_service(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(dnssd_service_freep) DnssdService *service = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *instance_name = NULL;
+        Manager *m = userdata;
+        DnssdService *s = NULL;
+        DnsTxtItem *last = NULL;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        if (m->mdns_support != RESOLVE_SUPPORT_YES)
+                return sd_bus_error_set_errno(error, EPROTONOSUPPORT);
+
+        service = new0(DnssdService, 1);
+        if (!service)
+                return sd_bus_error_set_errno(error, ENOMEM);
+        service->manager = m;
+
+        r = sd_bus_message_read(message, "sssqqq", &service->name,
+                                &service->name_tpl,
+                                &service->type, &service->port,
+                                &service->priority, &service->weight);
+        if (r < 0)
+                return sd_bus_error_set_errno(error, -r);
+
+        s = hashmap_get(m->dnssd_services, service->name);
+        if (s)
+                return sd_bus_error_setf(error, BUS_ERROR_DNSSD_SERVICE_EXISTS, "DNS-SD service '%s' exists already", service->name);
+
+        r = dnssd_render_instance_name(service, &instance_name);
+        if (r < 0)
+                return sd_bus_error_set_errno(error, -r);
+
+        if (!dnssd_srv_type_is_valid(service->type))
+                return sd_bus_error_set_errno(error, EINVAL);
+
+        r = sd_bus_message_enter_container(message, SD_BUS_TYPE_ARRAY, "{ss}");
+        if (r < 0)
+                return sd_bus_error_set_errno(error, -r);
+
+        while ((r = sd_bus_message_enter_container(message, SD_BUS_TYPE_DICT_ENTRY, "ss")) > 0) {
```

Comment:
hmm, iirc then at least the value of the TXT RRs is Ok to be binary, no? if so, this should be "say" or "ayay" instead?

Reply:
> hmm, iirc then at least the value of the TXT RRs is Ok to be binary, no? if so, this should be "say" or "ayay" instead?

I was reluctant to make it "say" as it would be more cumbersome to add TXT RRs in scripts invoking `busctl`. How about putting "sv" here instead and then accepting "s" and "ay" variants? This way would be more in line with the `StandardInputText=` and `StandardInputData=` approach in config files I guess...

Reply:
i think a better way out is to add a support for registering services to systemd-resolve instead, and make people use that. or maybe add a new tool for this purpose, dunno, but i don't think designing for busctl's ease-to-use is a good way forward...

Reply:
Alright. Will proceed with "say" then.